### PR TITLE
introduces a two-step patching process

### PR DESCRIPTION
**Pull request by ThyWoof**
. 2 new flags to control status:
- Main.Loaded = true when game finishes GameManager.BindPostDatabase
- Main.Enabled = true when all other patches were applied and models were loaded
- IsSafeToLoad() is true at the moment but idea is be able to use this to determine if safe to patch the game or not

### CODE REVIEW

**Comment by ThyWoof** [comment]
I only left a comment on Harmony decoration at GameManagerPatcher. Willing to reduce conflict during merge. Please move anything that got merged on this file Postfix() and move over to Main.FinishLoading()

**Comment by ThyWoof** [comment]
and of course delete GameManagerPatcher.cs as it won't be required anymore

**Comment by ThyWoof** [comment]
image

Added a different mod ID on purpose with a PreLoad postfix.

File: SolastaCommunityExpansion/Main.cs [hunk 1]
```diff
@@ -27,15 +30,17 @@ internal static bool Load(UnityModManager.ModEntry modEntry)
...
                 Logger = modEntry.Logger;
                 Mod = new ModManager<Core, Settings>();
-                Mod.Enable(modEntry, assembly);
-
                 Menu = new MenuManager();
-                Menu.Enable(modEntry, assembly);
 
-                Translations.Load(MOD_FOLDER);
+                Menu.Enable(ModEntry, assembly);
+                harmony.Patch(methodGameManagerBindPostDatabase, postfix: new HarmonyMethod(methodMainFinishLoading));
```
Comment (ChrisPJohn) [inline SolastaCommunityExpansion/Main.cs:43]: What's the benefit of making this patch dynamic?

Is this just a proof of concept for how we can dynamically load patches for features?

Comment (ThyWoof) [reply SolastaCommunityExpansion/Main.cs:43]: I’m just playing with it for now. We might need to have better control of the patches at some point and good to now we can control order, decide to enable / disable, etc.

**Comment by ThyWoof** [comment]
closing this one as this approach can produce some crazy bugs in the end.

**Pull request closed without merging**